internal/rpc/jsonrpc/methods.go

```diff
@@ -2103,6 +2104,92 @@ func (s *Server) getTransaction(ctx context.Context, icmd interface{}) (interfac
 	return ret, nil
 }
 
+// getTxOut handles a gettxout request by returning details about an unspent
+// output. In SPV mode, details are only returned for transaction outputs that
+// are relevant to the wallet.
+// To match the behavior in RPC mode, (nil, nil) is returned if the transaction
+// output could not be found (never existed or was pruned) or is spent by another
+// transaction already in the main chain.  Mined transactions that are spent by
+// a mempool transaction are not affected by this.
+func (s *Server) getTxOut(ctx context.Context, icmd interface{}) (interface{}, error) {
+	cmd := icmd.(*types.GetTxOutCmd)
+	w, ok := s.walletLoader.LoadedWallet()
+	if !ok {
+		return nil, errUnloadedWallet
+	}
+
+	// Attempt RPC passthrough if connected to DCRD.
+	n, err := w.NetworkBackend()
+	if err != nil {
+		return nil, err
+	}
+	if rpc, ok := n.(*dcrd.RPC); ok {
+		var resp json.RawMessage
+		err := rpc.Call(ctx, "gettxout", &resp, cmd.Txid, cmd.Vout, cmd.Tree, cmd.IncludeMempool)
+		return resp, err
+	}
+
+	txHash, err := chainhash.NewHashFromStr(cmd.Txid)
+	if err != nil {
+		return nil, rpcError(dcrjson.ErrRPCDecodeHexString, err)
+	}
+
+	if cmd.Tree != wire.TxTreeRegular && cmd.Tree != wire.TxTreeStake {
+		return nil, rpcErrorf(dcrjson.ErrRPCInvalidParameter, "Tx tree must be regular or stake")
+	}
+
+	// Attempt to read the unspent txout info from wallet.
+	outpoint := wire.OutPoint{Hash: *txHash, Index: cmd.Vout, Tree: cmd.Tree}
+	utxo, err := w.UnspentOutput(ctx, outpoint, *cmd.IncludeMempool)
+	if err != nil && !errors.Is(err, errors.NotExist) {
+		return nil, err
+	}
+	if utxo == nil {
+		return nil, nil // output is spent or does not exist.
+	}
+
+	if utxo.Tree != cmd.Tree {
+		// Not an error because it is technically possible (though extremely unlikely)
+		// that the required tx (same hash, different tree) exists on the blockchain.
```

Comment:
Doesn't this just mean "not found" in the same way that specifying the wrong Vout when calling `UnspentOutput` can `return nil, nil`?  i.e. caller specified the wrong tree and it "wasn't found" even though txid and vout were found, but the input arguments were all valid, hence no error.

Reply:
Oh, sorry I see that tree is provided to `UnspentOutput` as well...

Reply:
A transaction output for one tree was asked for, and another tree was returned.  This would be a bug in the wallet package.

If you are going to check this, and it doesn't match, it is an error.

Alternatively, you can reasonably trust that the wallet package won't return bogus data to you like this, and remove this entire check.

Reply:
> ... It is possible (at least theoretically) that the node has an entry for the specified txhash and vout but for a different tree. 
> ...
> ... Perhaps @davecgh can (re-)elaborate on the motivation for looking up utxos with tree as well as hash and vout.

That is correct.  While exceedingly unlikely, it is entirely possible that a given transaction hash exists in both trees because they are independent.  Since the output index is almost always a small number, that part would be very likely to collide should the former collision happen.  Thus, the correct behavior needs to ensure the tree is correct in one way or another.  A couple of options are to either include the tree in the lookup as dcrd does or alternatively potentially return multiple entries from the lookup and arrange for the calling code to check the tree and filter any results that don't apply there.  The former is very much my preference as it is more efficient and less error prone.